src/compiler/builder.ts

```diff
@@ -85,7 +97,7 @@ namespace ts {
      * State to store the changed files, affected files and cache semantic diagnostics
      */
     // TODO: GH#18217 Properties of this interface are frequently asserted to be defined.
-    export interface BuilderProgramState extends BuilderState {
+    export interface BuilderProgramState extends BuilderState, ReusableBuilderProgramState {

```

Comment:
Unified the structures here to make sure that all the properties in reusable builder state are correctly reflected.